coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,386 @@
+from coalib.bearlib.aspects import Root, Taste
+
+
+@Root.subaspect
+class Formatting:
+    """
+    Formatting refers to the visual appearance of source code.
+    """
+    class docs:
+        example = """
+        # Here is an example of python code with lots of
+        # formatting issues including: trailing spaces, missing spaces
+        # around operators, strange and inconsistent indentation etc.
+
+        z = 'hello'+'world'
+             def f ( a):
+                pass
+        """
+        example_language = 'python'
+        importance_reason = """
+        A coding style (the of rules or guidelines used when writing the
+        source code) can drastically affect the readability, and
+        maintainability of a program and might as well introduce bugs.
+        """
+        fix_suggestions = """
+        Defining a clearly and thoughtful coding style (based on the available
+        ones given the programming language in use) and strictly respect it or
+        apply it through out the implementation of a project.
+        """
+
+
+@Formatting.subaspect
+class Length:
+    """
+    The `Length` here refers to each line's length and the file length.
+    """
+    class docs:
+        example = """
+        # We assume that the maximum number of characters per line is 10
+        # and that the maximum number of lines per files is 3.
+
+        def run(bear, file, filename, aspectlist):
+            return bear.run(file, filename, aspectlist)
+        """
+        example_language = 'python'
+        importance_reason = """
+        Too long lines of code and too large files result in code difficult to
+        read, understand and maintain.
+        """
+        fix_suggestions = """
+        Length issues can be fixed by writing shorter lines of code(splitting
+        long lines into multiple shorter lines); writing shorter files(
+        splitting files into modules, writing shorter methods and classes.).
+        """
+
+
+@Length.subaspect
+class LineLength:
+    """
+    A line's length refers to the number of character found in it.
+    """
+    class docs:
+        example = """
+        print('The length of this line is 38')
+        """
+        example_langague = 'python'
+        importance_reason = """
+        Too long lines make code very difficult to read and maintain.
+        """
+        fix_suggestions = """
+        Splitting long lines of code into multiple shorter lines whenever
+        possible. Avoiding the usage of in-line language specific constructs
+        whenever they results in too long lines.
+        """
+    max_line_length = Taste[int](
+        'Maximum number of character for a line.',
+        (79, 80, 100, 120, 160), default=80)
+
+
+@Length.subaspect
+class FileLength:
+    """
+    A file's length refers to number of character found in it.
```

Comment:
Inconsistency! Here you refer only to number of characters, below only to number of lines